deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ParserUtility.java

```diff
@@ -234,17 +226,7 @@ private static void validateMapInternal(Map<String, Object> map, int level, int
             if((value != null) && (value instanceof Map))
             {
                 /* Codes_SRS_PARSER_UTILITY_21_052: [The validateMap shall throws IllegalArgumentException if the provided map contains more than maxLevel levels and those extra levels contain more than just metadata.] */
-                if(level <= maxLevel)
-                {
-                    validateMapInternal((Map<String, Object>) value, level, maxLevel, allowMetadata);
-                }
-                else
-                {
-                    if (!mapOnlyContainsMetaData((Map<String, Object>)value))
-                    {
-                        throw new IllegalArgumentException("Map exceed maximum of " + maxLevel + " levels");
-                    }
-                }
+                validateMapInternal((Map<String, Object>) value, allowMetadata);
```

Comment:
is the allow metadata still being used for any validation?

Reply:
there are other validations still being done; valid characters, size etc. This might be a good time to scrub through those and remove them.

Reply:
There are still a few validations on things like that which we could get rid of, sure. I'll take those out too.